sql/src/test/java/io/crate/analyze/PrivilegesDCLAnalyzerTest.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.analyze;
+
+import io.crate.action.sql.Option;
+import io.crate.action.sql.SessionContext;
+import io.crate.analyze.user.Privilege.Type;
+import io.crate.operation.user.User;
+import io.crate.operation.user.UserManager;
+import io.crate.operation.user.UserManagerProvider;
+import io.crate.sql.parser.SqlParser;
+import io.crate.test.integration.CrateDummyClusterServiceUnitTest;
+import io.crate.testing.SQLExecutor;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Collections;
+
+import static io.crate.analyze.user.Privilege.Type.*;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.isIn;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.*;
+
+public class PrivilegesDCLAnalyzerTest extends CrateDummyClusterServiceUnitTest {
+
+    private static final User GRANTOR_TEST_USER = new User("test", Collections.emptySet(), Collections.emptySet());
+
+    private SQLExecutor e;
+    private UserManagerProvider userManagerProvider;
+    private UserManager userManager;
+
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+        userManagerProvider = mock(UserManagerProvider.class);
```

Comment:
I think we could change the dependency in the Analyzer to only require a `Provider<UserManager>` and then it would be possible to use a simple `() -> userManager` below - getting rid of the mock here